### PR TITLE
add test for empty patch file

### PR DESCRIPTION
**Pull request by kbhawkey**
See issue #1060

### CODE REVIEW

File: pkg/patch/transformer/patchjson6902json_test.go [hunk 1]
```diff
@@ -163,3 +163,40 @@ func TestJsonPatchJSONTransformer_UnHappyTransform(t *testing.T) {
...
+	operations := []byte(``)
+
+	jpt, err := newPatchJson6902JSONTransformer(id, operations)
+
+	if err == nil {
+		t.Fatalf("unexpected error : %v", err)
+
+		if jpt != nil {
+			err = jpt.Transform(base)
+		}
```
Comment (Liujingfang1) [inline pkg/patch/transformer/patchjson6902json_test.go]: I don't think you need these lines. 
```
		if jpt != nil {
			err = jpt.Transform(base)
		}
```
The `jpt` is expected to be `nil`.

File: pkg/patch/transformer/patchjson6902json_test.go [hunk 2]
```diff
@@ -163,3 +163,40 @@ func TestJsonPatchJSONTransformer_UnHappyTransform(t *testing.T) {
...
+	}
+
+	if err != nil {
+		if !strings.HasPrefix(err.Error(), "json patch file is empty") {
+			t.Fatalf("expected error didn't happen, but got %v", err)
```
Comment (Liujingfang1) [inline pkg/patch/transformer/patchjson6902json_test.go]: Change it to:
```
t.Fatalf("expected %s, but got %v", "json patch file is empty", err)
```

File: pkg/patch/transformer/patchjson6902json_test.go [hunk 3]
```diff
@@ -163,3 +163,20 @@ func TestJsonPatchJSONTransformer_UnHappyTransform(t *testing.T) {
...
 }
+
+func TestJsonPatchJSONTransformer_EmptyPatchFile(t *testing.T) {
+	id := resid.NewResId(deploy, "deploy1")
+	operations := []byte(``)
+
+	_, err := newPatchJson6902JSONTransformer(id, operations)
+
+	if err == nil {
+		t.Fatalf("unexpected error : %v", err)
```
Comment (Liujingfang1) [inline pkg/patch/transformer/patchjson6902json_test.go]: change this line to
```
t.Fatalf("expected an error")
```
